### DISCUSSION TITLE
So2Sat getitem TypeError: img should be PIL Image. Got <class 'dict'>

### DISCUSSION THREAD

**Discussion by Akshatha-Mohan** [torchgeo/torchgeo#1293] [Q&A] [answered]
```
`import pdb
import PIL
import numpy as np
from torch.utils.data import Dataset
from torchgeo.datasets import So2Sat

class So2sat_Index(Dataset):
    def __init__(self, root, split='train',transform=None):
        self.root = root
        self.transform = transform
        self.split = split
        self.images = torchgeo.datasets.So2Sat(root, split = "train" ,transforms = transform, bands = So2Sat.BAND_SETS["s1"])       
        self.classes = self.images.classes
       
    def __getitem__(self, index):
        pdb.set_trace()
        sample = self.images.__getitem__(index)
        image = sample["image"]
        label = sample["label"]
        
        if self.transform is not None:
            image = self.transform(image)
        return index, image, label

    def __len__(self):
        return len(self.images)`
```

Hi,
I am trying to retrieve the index, image, and label that have undergone transformations in the getitem method of the So2Sat_Index class.
The error I'm facing is.
Error:
self.images.__getitem__(index)
*** TypeError: img should be PIL Image. Got <class 'dict'> How to rectify this?

**Comment by isaaccorley** [comment]
Can you post what transforms you are using? You can likely fix this by setting transforms=None when you define self.images

**Reply by Akshatha-Mohan** [reply]
'train': transforms.Compose([
            transforms.Resize(Network_parameters['resize_size']),
            transforms.ToTensor(),
            transforms.Normalize(mean, std),
            ]),

        'test': transforms.Compose([
                transforms.Resize(Network_parameters['resize_size']),
                transforms.ToTensor(),
                transforms.Normalize(mean, std),
            ]),
        }

**Reply by adamjstewart** [reply]
TorchGeo doesn't support torchvision transforms, they use a completely different format. You should instead use Kornia transforms. See https://torchgeo.readthedocs.io/en/stable/tutorials/transforms.html for a tutorial.

**Reply by Akshatha-Mohan** [reply]
Thank you. I'll check it out.

**Comment by adamjstewart** [comment]
What exactly are you trying to do? I don't think you need to make your own So2Sat_Index dataset, and it would be easier to create a subclass than to make a new dataset from scratch.

**Reply by Akshatha-Mohan** [reply]
My primary objective for implementing __getitem__ is to return the transformed image, index, and label, which will be passed to the model for the forward pass during training.

**Reply by adamjstewart** [reply, marked as answer]
The simpler way of doing this would be:
```python
from torchgeo.datasets import So2Sat

class So2Sat_Index(So2Sat):
    def __getitem__(index):
        sample = super()[index]
        sample["index"] = index
        return sample
```

**Reply by Akshatha-Mohan** [reply]
Thank you. This works for me
